### PR TITLE
Fix screenshotting and allow for HWC windowsurface composition

### PR DESCRIPTION
**Pull request by sletta**

### CODE REVIEW

**Comment by giucam** [comment]
I don't think i understand all the details of this PR, however i don't see any obvious mistakes and it works correctly as far as i can see, so +1 from me.

File: rpm/lipstick-qt5.spec [hunk 1]
```diff
@@ -1,7 +1,7 @@
-# 
+#
 # Do NOT Edit the Auto-generated Part!
```
Comment (thp) [inline rpm/lipstick-qt5.spec]: Feel free to just remove the spectacle comments altogether in the file, and the .yaml file if it hasn't been removed yet.

File: src/compositor/hwcimage.cpp [hunk 2]
```diff
@@ -587,6 +600,11 @@ void HwcImageTexture::bind()
     }
 }
 
+void hwcimage_delete_texture(void *, void *callbackData)
+{
+    delete (QObject *) callbackData;
```
Comment (thp) [inline src/compositor/hwcimage.cpp]: `delete static_cast<QObject *>(callbackData);` maybe?

Comment (thp) [inline src/compositor/hwcimage.cpp]: And if it's called from another thread, maybe even `static_cast<QObject *>(callbackData)->deleteLater();`?

File: src/compositor/lipstickcompositorwindow.h [hunk 3]
```diff
@@ -100,11 +106,11 @@ private slots:
     bool m_removePosted:1;
     bool m_mouseRegionValid:1;
     bool m_interceptingTouch:1;
+    bool m_mapped : 1;
```
Comment (thp) [inline src/compositor/lipstickcompositorwindow.h]: Maybe `bool m_mapped:1;` to be consistent with the bitfield lines above it.

**Comment by thp** [comment]
Added some minor comments, other than that looks good.

**Pull request closed without merging**